include/nuttx/virtio/virtio.h

```diff
@@ -0,0 +1,169 @@
+/****************************************************************************
+ * include/nuttx/virtio/virtio.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_NUTTX_VIRTIO_VIRTIO_H
+#define __INCLUDE_NUTTX_VIRTIO_VIRTIO_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <arch/spinlock.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* See virtio-v1.1-csprd01.pdf */
+
+#define VIRTIO_MAGIC    0x74726976
+#define VIRTIO_VERSION  0x2    /* NOTE: Legacy devices used 0x1 */
+
+#define VIRTIO_DEV_NET  0x1
+
+#define VIRTIO_STATUS_ACKNOWLEDGE   (1)
+#define VIRTIO_STATUS_DRIVER        (2)
+#define VIRTIO_STATUS_DRIVER_OK     (4)
+#define VIRTIO_STATUS_FEATURES_OK   (8)
+
+#define VIRTQ_DESC_F_NEXT  1   /* marks a buffer as continuing */
+#define VIRTQ_DESC_F_WRITE 2   /* marks a buffer as device write-only */
+
+#define mb() SP_DMB()
+
+#define getreg32(a)         (*(volatile uint32_t *)(a))
```

Comment:
should we add virtio_ prefix all mb/getreg32/putreg32 to avoid the potential conflict?

Reply:
done